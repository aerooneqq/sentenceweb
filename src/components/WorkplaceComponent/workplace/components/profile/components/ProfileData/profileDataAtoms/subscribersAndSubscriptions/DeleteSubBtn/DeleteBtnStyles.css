.deleteSubBtn{ 
    width: 80px;
    height: 30px;

    border-width: 1px;
    border-color: red;
    border-style: solid;
    border-radius: 7px;

    font-family: Arial;
    color: white;
    font-size: 13px;

    background: red;
}

.deleteSubBtn:hover{ 
    background: linear-gradient(-120deg, #ee0000, red);
}

.deleteSubBtn:focus{ 
    outline: none;
}