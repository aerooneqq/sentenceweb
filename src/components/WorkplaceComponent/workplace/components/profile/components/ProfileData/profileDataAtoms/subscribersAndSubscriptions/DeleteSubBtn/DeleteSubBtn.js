import React, {Component} from "react";

//Styles
import "./DeleteBtnStyles.css";

/**
 * PROPS LIST:
 * 1) deleteSubscriber
 * 2) userID
 */
export default class DeleteSubBtn extends Component { 
    constructor(props) { 
        super(props);
        this.handleBtnClick = this.handleBtnClick.bind(this);
    }

    handleBtnClick() { 
        this.props.deleteSubscriber(this.props.userID);
    }

    render() { 
        return( 
            <button className = "deleteSubBtn"
                    onClick = {this.handleBtnClick}>
                Delete
            </button>
        )
    }
}