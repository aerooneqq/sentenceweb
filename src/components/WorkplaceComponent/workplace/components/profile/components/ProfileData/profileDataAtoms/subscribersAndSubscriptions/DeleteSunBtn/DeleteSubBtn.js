import React, {Component} from "react";

//Styles
import "./DeleteBtnStyles.css";

export default class DeleteSubBtn extends Component{ 
    constructor(props){ 
        super(props);
        this.handleBtnClick = this.handleBtnClick.bind(this);

        alert(this.props.userID);
    }

    handleBtnClick(){ 
        this.props.deleteSubscriber(this.props.userID);
    }

    render(){ 
        return( 
            <button class = "deleteSubBtn"
                    onClick = {this.handleBtnClick}>
                Delete
            </button>
        )
    }
}