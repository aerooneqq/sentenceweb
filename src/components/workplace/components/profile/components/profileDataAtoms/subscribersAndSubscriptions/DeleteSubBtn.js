import React, {Component} from "react";

//Styles
import "./styles/DeleteBtnStyles.css";

export default class DeleteSubBtn extends Component{ 
    constructor(props){ 
        super(props);
    }

    render(){ 
        return( 
            <button class = "deleteSubBtn">
                Delete
            </button>
        )
    }
}