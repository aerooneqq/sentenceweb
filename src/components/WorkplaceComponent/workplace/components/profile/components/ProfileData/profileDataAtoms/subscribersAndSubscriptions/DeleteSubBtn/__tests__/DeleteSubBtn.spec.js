import React from "react"; 
import {shallow} from "enzyme"; 
import {shallowToJson} from "enzyme-to-json";
import DeleteSubBtn from "../DeleteSubBtn";

const spy = jest.fn((x) => {});
const props = { 
    deleteSubscriber: spy
}

describe("DeleteSubBtn", () => { 
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<DeleteSubBtn />))).toMatchSnapshot();
    });

    it("should call deleteSubscriber from props when clicked", () => { 
        const deleteSubBtn = shallow(<DeleteSubBtn {...props} />);

        deleteSubBtn.find("button.deleteSubBtn").simulate("click");

        expect(spy).toBeCalled();
    });
});