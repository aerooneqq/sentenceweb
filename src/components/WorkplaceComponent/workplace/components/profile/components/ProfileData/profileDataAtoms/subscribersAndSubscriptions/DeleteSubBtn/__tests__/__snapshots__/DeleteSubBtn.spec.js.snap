// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteSubBtn should render correctly 1`] = `
<button
  className="deleteSubBtn"
  onClick={[Function]}
>
  Delete
</button>
`;
